oauth2_http/java/com/google/auth/oauth2/UserCredentials.java

```diff
@@ -129,7 +129,49 @@ public AccessToken refreshAccessToken() throws IOException {
     return access;
   }
 
+  /**
+   * Returns client ID of the credential from the console.
+   *
+   * @return client ID
+   */
+  public final String getClientId() {
+    return clientId;
+  }
+
+  /**
+   * Returns client secret of the credential from the console.
+   *
+   * @return client secret
+   */
+  public final String getClientSecret() {
+    return clientSecret;
+  }
+
+  /**
+   * Returns the refresh token resulting from a OAuth2 consent flow.
+   *
+   * @return refresh token
+   */
   public final String getRefreshToken() {
     return refreshToken;
   }
+
+  /**
+   * Returns HTTP object used to get access tokens.
+   *
+   * @return HTTP transport
+   */
+  public final HttpTransport getHttpTransport() {
+    return transport;
+  }
+
```

Comment:
This is for rarer extensibility and testing scenarios. I would not have this as a public property.